Source/SwiftLintFramework/Rules/VerticalWhitespaceClosingBracesRule.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+import SourceKittenFramework
+
+private extension File {
+    func violatingRanges(for pattern: String) -> [NSRange] {
+        return match(pattern: pattern, excludingSyntaxKinds: SyntaxKind.commentAndStringKinds)
+    }
+}
+
+public struct VerticalWhitespaceClosingBracesRule {
+    public init() {}
+
+    private static let violatingToValidExamples: [String: String] = [
+        "    print(\"x is 5\")\n\n}": "    print(\"x is 5\")\n}",
+        "    print(\"x is 5\")\n\n\n}": "    print(\"x is 5\")\n}",
+        "    print(\"x is 5\")\n    \n}": "    print(\"x is 5\")\n}"
+    ]
+
+    private let pattern = "((?:\\n[ \\t]*)+)(\\n[ \\t]*[)}\\]])"
+}
+
+extension VerticalWhitespaceClosingBracesRule: OptInRule {
+    public var configurationDescription: String { return "N/A" }
+
+    public init(configuration: Any) throws {}
+
+    public static let description = RuleDescription(
+        identifier: "vertical_whitespace_closing_braces",
+        name: "Vertical Whitespace before Closing Braces",
+        description: "Don't include vertical whitespace (empty line) before closing braces.",
+        kind: .style,
+        nonTriggeringExamples: Array(Set(violatingToValidExamples.values)),
```

Comment:
usually single-line comments and comment blocks are added as nonTriggeringExamples

Reply:
also this is a line that includes a closing brace but should accept vertical whitespace before it:
```swift
let a = b.map { $0.foo }
```

Reply:
I added the following example:
```swift
class Name {
    run(5) { x in print(x) }

}
```

I hope this covers the case you mentioned.